### PR TITLE
issues: implement functionality to add images to post

### LINKED ISSUE #51: Implement a module for posting images along with an issue.
We can discuss the constraints for the same here, For ex : the max size of file, file ext, or the max number of images allowed. 

Basic idea would be 
1. to have a nodejs static image server, which only allows access to our mobile and web clients (we can add this later, just a few lines of code)
2. Hashing the `name of image and date` and storing this uniquely in the mongoDB as the filename
3. Using this filename and storing this on a folder in `localstorage` (we can later even shift to a web based file storage for scaling if needed, but mostly won't need to).
4. Sending in the `url` to the image resource in `json` back to the clients that are requesting it and, then the client will try to download and/or access the image(s) from that url.

### PR DESCRIPTION
**Pull request by FaiyazMujawar**
Fixes #51 

In this PR,

- Implemented a separate route for posting images.
- Implemented an Image Mongoose Schema, for storing each image meta data for logging purpose.
- Implemented functionality to add images to post.
- Updated documentation accordingly.

### CODE REVIEW

File: controllers/IssueController.js [hunk 1]
```diff
@@ -398,6 +442,7 @@ const validateData = (title, description, section, scope) => {
       .regex(/\w+/)
       .rule({ message: "Please provide a valid description" })
       .required(),
+    images: Joi.array().max(3).required(),
```
Comment (drigio) [inline controllers/IssueController.js:444]: Images arent `required` right?

File: models/Issue.js [hunk 2]
```diff
@@ -10,6 +10,10 @@ const IssueSchema = new Schema({
     type: String,
     default: "",
   },
+  images: {
+    type: [String],
+    required: true,
```
Comment (drigio) [inline models/Issue.js:15]: Do we have to make images `required`?

File: services/FileService.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+/** Maximum image count */
+const MAX_IMAGE_COUNT = 3;
+/** Destination for storing images */
+const IMAGE_DESTINATION = "./images";
+
+/** Multer Storage Engine, configured to store files locally */
+const storage = multer.diskStorage({
+  destination: IMAGE_DESTINATION,
+  filename: (_, { originalname }, cb) => {
+    cb(null, `${originalname}${extname(originalname)}`);
```
Comment (drigio) [inline services/FileService.js]: Do you think we should be naming it this way? Cuz if there are multiple images with the same name, they might get overridden, which can be a possibility during multiple requests.

File: services/FileService.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      }
+    );
+
+    const { data, status } = response;
+    if (status === 200) {
+      return data;
+    }
+    return null;
+  } catch (error) {
+    throw error;
```
Comment (drigio) [inline services/FileService.js:56]: throwing error, if not handled later might crash the system. Are you catching it at a later stage?

File: controllers/IssueController.js [hunk 5]
```diff
@@ -130,18 +171,21 @@ const addIssue = async (req, res) => {
         title,
         description,
         section,
+        images,
         department,
         scope,
         id
       );
+      console.log("issue created");
+      await updateImageIssueId(images, id);
```
Comment (drigio) [inline controllers/IssueController.js]: `images` can be null if no image is uploaded for an issue and it might give errors.

**Review by drigio — CHANGES_REQUESTED** [review]

File: controllers/IssueController.js (hunk 1 above)
Comment (FaiyazMujawar) [reply controllers/IssueController.js:444]: They are required. Even if there are jo images, an empty array must be passed to maintain consistency in database schema

File: services/FileService.js (hunk 3 above)
Comment (FaiyazMujawar) [reply services/FileService.js]: It was just to time being. As u earlier said, it would be hashed

File: controllers/IssueController.js (hunk 5 above)
Comment (FaiyazMujawar) [reply controllers/IssueController.js]: Nope. $in would not work if there are no matches. Hence I made images as required.

File: services/FileService.js (hunk 3 above)
Comment (drigio) [reply services/FileService.js]: Make that change then before we merge in.

File: controllers/IssueController.js (hunk 1 above)
Comment (drigio) [reply controllers/IssueController.js:444]: Okay

File: controllers/IssueController.js (hunk 5 above)
Comment (drigio) [reply controllers/IssueController.js]: If there are no images we shouldn't call this method in the first place.

Comment (FaiyazMujawar) [reply controllers/IssueController.js]: Okay. Can be checked at the time of issue creation.

**Review by drigio — APPROVED** [review]
LGTM! I'll merge in once you give me a head's up. Also I hope you have seen the new code working?

**Pull request merged**